services/compute/compute.go

```diff
@@ -1160,3 +1160,12 @@ type ProxyConfiguration struct {
 	// Alternative CA cert to use for connecting to proxy server
 	TrustedCa *string `json:"trustedca,omitempty"`
 }
+
+type VirtualMachineDiscovery struct {
```

Comment:
As we discussed yesterday, I think we don't need this struct after all as we're just using the `VirtualMachine` struct

Reply:
@mduppre I think we will need this structure if we want to change the API function signature.

Reply:
yeah, you're right, I wrote this comment before we chatted about the API function signature. So let's leave it and I'll look into converting the objects